File: src/_1ProgrammingBasicsWithJavaMay2023/_03ConditionalStatementsAdvanced/_2Exercise/_05Journey.java
package _1ProgrammingBasicsWithJavaMay2023._03ConditionalStatementsAdvanced._2Exercise;

import java.util.Scanner;

public class _05Journey {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double budget = Double.parseDouble(scanner.nextLine());
        String season = scanner.nextLine();
        String destination = "";
        String place = "";
        double usedMoneyFromBudget = 0.0;
        if (budget <= 100) {
            destination = "Bulgaria";
            if (season.equals("summer")) {
                place = "Camp";
                usedMoneyFromBudget = budget * 0.30;
            } else if (season.equals("winter")) {
                place = "Hotel";
                usedMoneyFromBudget = budget * 0.70;
            }
        } else if (budget > 100 && budget <= 1000) {
            destination = "Balkans";
            if (season.equals("summer")) {
                place = "Camp";
                usedMoneyFromBudget = budget * 0.40;
            } else if (season.equals("winter")) {
                place = "Hotel";
                usedMoneyFromBudget = budget * 0.80;
            }
        } else if (budget > 1000) {
            destination = "Europe";
            if (season.equals("summer")) {
                place = "Hotel";
                usedMoneyFromBudget = budget * 0.90;
            } else if (season.equals("winter")) {
                place = "Hotel";
                usedMoneyFromBudget = budget * 0.90;
            }
        }
        System.out.printf("Somewhere in %s%n%s - %.2f", destination, place, usedMoneyFromBudget);
    }
}

File: src/_1ProgrammingBasicsWithJavaScriptNovember2024/_3ConditionalStatementsAdvanced/_2Exercise/_05Journey.js
function journey(budged, season) {
    const PERCENT_USE_MONEY_FROM_BUDGED_IN_BULGARIA_SUMMER_TIME = 0.30;
    const PERCENT_USE_MONEY_FROM_BUDGED_IN_BULGARIA_WINTER_TIME = 0.70;
    const PERCENT_USE_MONEY_FROM_BUDGED_IN_BALKANS_SUMMER_TIME = 0.40;
    const PERCENT_USE_MONEY_FROM_BUDGED_IN_BALKANS_WINTER_TIME = 0.80;
    const PERCENT_USE_MONEY_FROM_BUDGED_IN_EUROPA_ALL_TIME = 0.90;

    let destination = "";
    let placeToStay = "";
    let moneyUse = 0.0;
    if (budged <= 100) {
        destination = "Bulgaria";
        if (season === "summer") {
            placeToStay = "Camp";
            moneyUse = budged * PERCENT_USE_MONEY_FROM_BUDGED_IN_BULGARIA_SUMMER_TIME;
        } else if (season === "winter") {
            placeToStay = "Hotel";
            moneyUse = budged * PERCENT_USE_MONEY_FROM_BUDGED_IN_BULGARIA_WINTER_TIME;
        }
    } else if (budged <= 1000) {
        destination = "Balkans";
        if (season === "summer") {
            placeToStay = "Camp";
            moneyUse = budged * PERCENT_USE_MONEY_FROM_BUDGED_IN_BALKANS_SUMMER_TIME;
        } else if (season === "winter") {
            placeToStay = "Hotel";
            moneyUse = budged * PERCENT_USE_MONEY_FROM_BUDGED_IN_BALKANS_WINTER_TIME;
        }
    } else if (budged > 1000) {
        destination = "Europe";
        placeToStay = "Hotel";
        moneyUse = budged * PERCENT_USE_MONEY_FROM_BUDGED_IN_EUROPA_ALL_TIME;
    }

    console.log(`Somewhere in ${destination}`);
    console.log(`${placeToStay} - ${moneyUse.toFixed(2)}`)
}